src/validation/rules/LoneSchemaDefinition.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import type { SDLValidationContext } from '../ValidationContext';
+import { GraphQLError } from '../../error';
+import type { ASTVisitor } from '../../language/visitor';
+
+export function schemaDefinitionNotAloneMessage(): string {
+  return 'Must provide only one schema definition.';
+}
+
+export function canNotDefineSchemaWithinExtension(): string {
+  return 'Cannot define a new schema within a schema extension.';
+}
+
+/**
+ * Lone Schema definition
+ *
+ * A GraphQL document is only valid if it contains only one schema definition.
+ */
+export function LoneSchemaDefinition(
+  context: SDLValidationContext,
+): ASTVisitor {
+  const oldSchema = context.getSchema();
+  const alreadyDefined =
+    oldSchema &&
+    (oldSchema.astNode ||
+      oldSchema.getQueryType() ||
+      oldSchema.getMutationType() ||
+      oldSchema.getSubscriptionType());
+
+  const schemaNodes = [];
+  return {
+    SchemaDefinition(node) {
+      if (alreadyDefined) {
+        context.reportError(
+          new GraphQLError(canNotDefineSchemaWithinExtension(), [node]),
```

Comment:
It might be nice to point back to the originally defined schema if provided. But I could be wrong here and that would just be noise.

Also a very small nit provided you're only going to use this node in the error, not sure it's worth changing: 
```
new GraphQLError(canNotDefineSchemaWithinExtension(), node),
```

Reply:
@mjmahone I was thinking that in some systems (e.g. GraphQL is very popular for CMS) clients can provide their own extension SDL. So was concern that we can leak some internal data in printed errors.

Thinking more about this I think I'm too paranoid since `validateSchema` already leaking this info so if someone decides he doesn't want origin node to be printed he needs to sanitize `GraphQLError`s anyway. Plus such complex systems almost certainly using `GraphQL*Type` classes anyway.